### PR TITLE
chore(executor): upgrade `kubectl` to 1.18. Closes #3996

### LINKED ISSUE #3996: Old version of kubectl 1.15, latest stable is 1.19.1
## Summary 

The Dockerfile for argo is pulling in [1.15 of kubectl](https://github.com/argoproj/argo/blob/a37d0a729c206040463d41e88e09814c1bf622dd/Dockerfile#L53). Given the comment in the Dockerfile, I expected this to be 1.19.1 or at least 1.18.8.

The Kubernetes project maintains release branches for the most recent three minor releases (1.19, 1.18, 1.17).
Kubernetes 1.19 and newer receive approximately 1 year of patch support.
Kubernetes 1.18 and older received approximately 9 months of patch support.
kubectl is supported within one minor version (older or newer) of kube-apiserver.

Example: kube-apiserver is at 1.18, then kubectl is supported at 1.19, 1.18, and 1.17

Given the above, argo should likely remain on one minor less (1.18) than the current stable (1.19). This allows support for all three active releases of k8s.

## Diagnostics

I reviewed the master branch and latest release branch of argo.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by ricochet**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or **(c) this is a chore**.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

I've used https://github.com/GoogleContainerTools/container-structure-test for docker specific tests. I can add this if that's desirable.

I've signed the CNCF CLA. Is there another one?

### CODE REVIEW

File: USERS.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ Currently, the following organizations are **officially** using Argo Workflows:
 1. [Sage (Sage AI Labs)](https://sage.com/)
 1. [SAP Fieldglass](https://www.fieldglass.com/)
 1. [SAP Hybris](https://cx.sap.com/)
+1. [SAS Institute Inc.](https://www.sas.com/)
```
Comment (simster7) [inline USERS.md:75]: 🎉

File: Dockerfile [hunk 2]
```diff
@@ -50,8 +50,8 @@ FROM debian:10.3-slim as argoexec-base
 
 ARG IMAGE_OS=linux
 
-# NOTE: keep the version synced with https://storage.googleapis.com/kubernetes-release/release/stable.txt
-ENV KUBECTL_VERSION=1.15.1
+# NOTE: kubectl version should be one minor version less than https://storage.googleapis.com/kubernetes-release/release/stable.txt
```
Comment (simster7) [inline Dockerfile:53]: Why one minor version less?

**Review by simster7 — COMMENTED** [review]
> I've signed the CNCF CLA. Is there another one?

Yes, we have our own CLA. Please see the comment from CLAAssistant

File: Dockerfile (hunk 2 above)
Comment (ricochet) [reply Dockerfile:53]: The Kubernetes project maintains release branches for the most recent three minor releases (1.19, 1.18, 1.17).
Kubernetes 1.19 and newer receive approximately 1 year of patch support.
Kubernetes 1.18 and older received approximately 9 months of patch support.
kubectl is supported within one minor version (older or newer) of kube-apiserver.

Example: kube-apiserver is at 1.18, then kubectl is supported at 1.19, 1.18, and 1.17

Given the above, argo should likely remain on one minor less (1.18) than the current stable (1.19). This allows support for all three active releases of k8s.

**Comment by simster7** [comment]
Awesome, thanks. Please sign the CLA following this comment: https://github.com/argoproj/argo/pull/3999#issuecomment-690652734

Then we should be good to go

**Comment by ricochet** [comment]
> Awesome, thanks. Please sign the CLA following this comment: [#3999 (comment)](https://github.com/argoproj/argo/pull/3999#issuecomment-690652734)
> 
> Then we should be good to go

I'm waiting on legal. I should have the CLA signed sometime next week.

**Review by simster7 — APPROVED** [review]
Thanks @ricochet!

**Pull request merged**